WriteHDLUtils.py

```diff
@@ -828,15 +834,35 @@ def writeModuleInst_generic(module, hls_src_dir, f_writeTemplatePars,
                     foundMatch = f_matchArgPortNames(argname, portname)
 
                 if foundMatch:
+                    # Create temporary argument name as argname can be an array and have several matches
+                    tmp_argname = argname
+                    argname_is_array = (tmp_argname.find('[') != -1) # Check if array
+
+                    # Special case if argname is an array
+                    if argname_is_array:
+                        tmp_argname = tmp_argname.split('[')[0] # Remove "[...]"
+                        # Keep track of the array names and the number of array elements
+                        if tmp_argname in array_dict:
+                            array_dict[tmp_argname] += 1
```

Comment:
Does this eliminate the need to know in advance the length of the array of memories that is passed to the templated function? That's great if so.

Reply:
Not sure I understand this... It doesn't check the length of the array in the actual HLS, instead it just assumes that the number of "array elements" that ends up having ports are the same as in the wiring. 

In the Matchcalculator, the size of the array for the fullmatch memories is set to 8, but only 2 memories are used so the HLS ends up only creating those two ports. So looking at the array length wouldn't really work anyway?
https://github.com/cms-L1TK/firmware-hls/blob/master/TrackletAlgorithm/MatchCalculatorTop.h#L7

Reply:
So it sounds like there never was a need to know in advance how long the memory arrays are. I had this vague worry that it was a problem that needed solving, but it seems not. This is good news anyways.